### ISSUE TITLE
AUTO_STOP + UserConnected can fail to produce good optimisations

### ISSUE THREAD

**Issue by dannyfallon** [awslabs/workspaces-cost-optimizer#17]
AWS Workspaces running in AUTO_STOP running mode are billed per hour they are turned on, not per hour of `UserConnected`. The workspace does not shut down immediately when `UserConnected` is false - the minimum idle time is 1 hour, but can be up to 48 hours.

If somebody uses a workspace for 2hrs 59min a day, here is the comparison of what the current Cost Optimiser computes as billable and the actual AWS billable cost across two situations and two values of idle timeout:

Idle (h) |Connected|UserConnected (h)| Cost Optimiser (h) | Actual (h)
|---|---|---|---|---
1|09:00-11:59|3|3|4
1|09:15-12:14|4|4|5
4|09:00-11:59|3|3|7
4|09:15-12:14|4|4|8

Going with the 09:00-11:59 example, the cost optimiser will never upgrade even if the pattern repeated for an entire month. On a 30 day month this will result in an actual bill of 120h for that workspace with a 1h timeout, and an actual bill of 210h with a 4h timeout. Both of these actual billable hours are far above the limits for every workspace type except `Graphics`. It might be higher still since I've not taken into account any maintenance windows where the workspace is powered on and I'm fairly sure that is billable.

I can come up with a few more scenarios where `UserConnected` is problematic - people who connect multiple times per day for short periods to queue up work, people running cloud developer environments on workspaces who use high idle timeouts because they're running dev servers on them etc.

Because AWS Workspaces does not have a metric for powered on (aka billable), I think the only true and accurate representation of billable hours when in AUTO_STOP mode is the number of hours the instance is not `STOPPED` and prior to 2.2.1 this is what the computation was done with.

I'm hesitant to push a PR that restores the 2.2.0 behaviour since it was changed quite recently so I said I'd open this to explain instead and talk about it 😄

**Comment by rakshb** [comment]
Hello @dannyfallon  thanks for raising this issue. We went with UserConnected metric based on service team's recommendation (https://aws.amazon.com/workspaces/faqs/). 'UserConnected' metric is more accurate than 'Stopped' metric in most scenarios especially when workspaces are launched with default AutoStop time. 
For the use case you mentioned, we were able to reproduce the same behavior from our end and will be researching this and add it as a backlog item
